File: test/lang/1-pascal/fib_1.pas
library fibs;

Uses sysutils;


const
	version = 1;


function fibalgo (value: Int64) : Int64;
begin
  if value < 0
  then fibalgo := 0
  else if value < 2
  then fibalgo := value
  else fibalgo := fibalgo(value - 1) + fibalgo(value - 2)
end;


function fib (value: Int64) : Int64;
  cdecl; export;
begin
  fib := fibalgo(value)
end;

exports
  fib;


procedure printfib (value: Int64);
   cdecl; export;
var
  r: Int64;
  fmt,s : string;
begin
  r:=fibalgo(value);
  fmt:='[Pascal Fibonacci Library v%d] fib(%d) = %d';
  s:=Format(fmt,[version,value,r]);
  Writeln(s);
  Flush(Output);
end;

exports
  printfib;

end.

File: test/lang/1-cpp/fib_1.cpp
#include "fib.h"

#include <iostream>

const unsigned short version = 1;

static long fibalgo(long value) {
	if (value < 0)
		return 0;
	else if (value < 2)
		return value;
	else
		return fibalgo(value - 1) + fibalgo(value - 2);
}

long fib(long value) {
	return fibalgo(value);
}

void printfib(long value) {
	std::cout << "[C++ Fibonacci Library v" << version << "] fib(" << value << ") = " << fib(value) << std::endl << std::flush;
}